fuse_core/src/async_motion_model.cpp

```diff
@@ -64,15 +64,16 @@ bool AsyncMotionModel::apply(Transaction& transaction)
   auto callback = boost::make_shared<CallbackWrapper<bool>>(
     std::bind(&AsyncMotionModel::applyCallback, this, std::ref(transaction)));
   auto result = callback->getFuture();
-  callback_queue_.addCallback(callback);
+  callback_queue_.addCallback(callback, reinterpret_cast<uint64_t>(this));
```

Comment:
Just curious: the `reinterpret_cast<uint64_t>(this)` ID would be the same for all the callbacks added by `this` object. Is this intended. From the `callback_queue.cpp` it looks like it'll add all the callbacks and `removeByID` will remove all the callbacks with the given ID.

Reply:
Any callbacks added to the callback queue by ROS (e.g. message subscription callbacks, service callbacks, timer callbacks) are managed internally by ROS and each have their own ID number. All of the callbacks added to the callback queue "manually" are now assigned a known ID number, but it is the same for all of the callbacks. For the AsyncMotionModel, this will be calls to `applyCallback()`, `onGraphUpdate()`, `onStart()`, and `onStop()`. At the moment, I have no particular need to remove the callbacks from, say `onStart()`, separately from any of the others. If you have a use-case in mind where clearing individual callbacks types would be valuable, I can definitely assign different IDs to them. Though `onStart()` and `onStop()` should probably be treated as the same "unit".

Reply:
I don't have a use case for removing individual callbacks atm. It's fine as is. I just wanted to confirm that potential (but unlikely) limitation. Thanks. :smiley: